satip/eumetsat.py

```diff
@@ -577,12 +579,32 @@ def create_and_download_datatailor_data(
             datatailor = eumdac.DataTailor(eumdac.AccessToken((self.user_key, self.user_secret)))
 
             # sometimes the customisation fails first time, so we try twice
+            # This is from Data Tailor only allowing 3 customizations at once
+            # So this should then continue until it is created successfully
             try:
                 customisation = datatailor.new_customisation(dataset_id, chain=chain)
             except Exception:
-                log.debug("Did not customisation first time, so trying again after 2 seconds")
-                time.sleep(2)
-                customisation = datatailor.new_customisation(dataset_id, chain=chain)
+                log.debug("Customization not made successfully, so trying again after less than 3 customizations")
+                created_customization = False
+                while not created_customization:
+                    time.sleep(3)
+                    try:
+                        num_running_customizations = 0
+                        for customisation in datatailor.customisations:
+                            try:
```

Comment:
im a bit worried there are 3 try statements all together. This makes it hard to read, and im not sure on the logic if somthing fails. 

I also thin that `while` needs a timeout on it

Reply:
Okay, I've added a timeout and removed two of the tries